clients/spark/core/src/main/scala/io/treeverse/clients/GarbageCollector.scala

```diff
@@ -209,48 +229,76 @@ object GarbageCollector {
   ): Dataset[Row] = {
     val commitsDF = getCommitsDF(runID, commitDFLocation, spark)
     val rangesDF = getRangesDFFromCommits(commitsDF, repo, conf)
-    getExpiredEntriesFromRanges(rangesDF, conf, repo)
+    val expired = getExpiredEntriesFromRanges(rangesDF, conf, repo)
+
+    // validate deduplications
+    val activeRangesDF = rangesDF.where("!expired")
+    subtractDeduplications(expired, activeRangesDF, conf, repo, spark)
+  }
+
+  private def subtractDeduplications(expired: Dataset[Row], activeRangesDF: Dataset[Row], conf: APIConfigurations, repo: String, spark: SparkSession) : Dataset[Row]= {
+    val expiredAddr: Set[String] = expired.select("address").collect().map(_.getString(0)).toSet
+    val ranges : Seq[String] = activeRangesDF.select("range_id").collect().map(_.getString(0)).toSeq.distinct
+    val rangesRDD = spark.sparkContext.parallelize(ranges)
+
+    val activeAddresses =rangesRDD.flatMap(range=> {
+      getEntryAddressesIfInSet(range, conf, repo, expiredAddr)
+    }).map(x => Row(x))
+
+    val schema = new StructType().add(StructField("address",StringType, true))
+    val activeDF = spark.createDataFrame(activeAddresses, schema)
+    expired.join(
+      activeDF,
+      expired("address") === activeDF("address"),
+      "leftanti"
+    )
   }
 
   def main(args: Array[String]) {
     val spark = SparkSession.builder().getOrCreate()
-    if (args.length != 4) {
+    if (args.length != 2) {
       Console.err.println(
-        "Usage: ... <repo_name> <runID> s3://storageNamespace/prepared_commits_table s3://storageNamespace/output_destination_table"
+        "Usage: ... <repo_name> <region>"
```

Comment:
About the region - can we use [withForceGlobalBucketAccessEnabled](https://stackoverflow.com/a/46111903/2696779) to start the client without the region?

Reply:
Thats for the previous SDK version. seems like they didn't solve it yet for the current version.
https://github.com/aws/aws-sdk-java-v2/issues/2229
https://github.com/aws/aws-sdk-java-v2/issues/52